docs/guides/configure/session-tasks.mdx

```diff
@@ -64,130 +60,100 @@ If the prebuilt components don't meet your specific needs or if you require more
-
-const isProtectedRoute = createRouteMatcher(['/dashboard(.*)', '/forum(.*)'])
-
-export default clerkMiddleware(async (auth, req) => {
-  // pending users won't be able to access protected routes
-  // and will be redirected to the sign-in page
-  if (isProtectedRoute(req)) await auth.protect()
-})
+The `<RedirectToTasks />` control component redirects users to the appropriate task page when they have pending session tasks.
 
-export const config = {
-  matcher: [
-    // Skip Next.js internals and all static files, unless found in search params
-    '/((?!_next|[^?]*\\.(?:html?|css|js(?!on)|jpe?g|webp|png|gif|svg|ttf|woff2?|ico|csv|docx?|xlsx?|zip|webmanifest)).*)',
-    // Always run for API routes
-    '/(api|trpc)(.*)',
-  ],
-}
+```tsx
+<>
+  <SignedOut>
+    <RedirectToTasks />
+  </SignedOut>
+</>
 ```
 
-For `auth().isAuthenticated`, it would return `false` if the user has a `pending` session. Pending users will be redirected to the sign-in page, where the `<SignIn />` component will prompt them to fulfill the session tasks. Once finished, their session will move from `pending` to a `signed-in` state.
+### Middleware-based redirects
 
-```tsx {{ filename: 'app/middleware.ts', mark: [[6, 12]] }}
-import { clerkMiddleware, createRouteMatcher } from '@clerk/nextjs/server'
+You can use the `isAuthenticated` property in middleware to protect routes. By default, it will return `false` if the user has a `pending` session. Then, you can handle how to respond to pending users, such as redirecting them to the sign-in page using the `redirectToSignIn()` method.
 
+```tsx
 const isProtectedRoute = createRouteMatcher(['/dashboard(.*)', '/forum(.*)'])
 
 export default clerkMiddleware(async (auth, req) => {
   const { isAuthenticated, redirectToSignIn } = await auth()
 
   // `isAuthenticated` will be `false` for pending users
-  // and they will be redirected to the sign-in page
+  // and pending users won't be able to access protected routes
   if (!isAuthenticated && isProtectedRoute(req)) {
+    // Add logic to handle pending users
+    // This example redirects pending users to the sign-in page
+    // so they can fulfill the session tasks
     return redirectToSignIn()
   }
 })
-
-export const config = {
-  matcher: [
-    // Skip Next.js internals and all static files, unless found in search params
-    '/((?!_next|[^?]*\\.(?:html?|css|js(?!on)|jpe?g|webp|png|gif|svg|ttf|woff2?|ico|csv|docx?|xlsx?|zip|webmanifest)).*)',
-    // Always run for API routes
-    '/(api|trpc)(.*)',
-  ],
-}
 ```
 
-### Session handling
+<If sdk="nextjs">
+  For `auth.protect()`, signed-out users will be redirected to the sign-in page. In the following example, `pending` users will be redirected to the sign-in page, where the `<SignIn />` component will prompt them to fulfill the session tasks. Once finished, their session will move from `pending` to an `active` (signed-in) state.
+
+  ```tsx {{ filename: 'middleware.ts', mark: [[6, 8]] }}
```

Comment:
Does this need to be `proxy.ts` now? If true, do we need to update **Middleware** references to **Proxy** in this file too?